### PR TITLE
Update write_image signature

### PR DESCRIPTION
**Pull request by constantinpape**
This PR updates the signatures of `write_image`, `write_multiscale` and `write_multiscale_labels`:
- Removes `byte_order` from `write_image`: this was not used at all
- Removes `chunks` from all of them to avoid duplication with `storage_options`. 
- Moves `retuple` from `write_image` to `write_multiscale`, which is where the "actual" writing of data happens.

If we agree on these changes I will also add `write_labels` here (corresponding to `write_image`, but for image label data).

cc @will-moore @sbesson @joshmoore

### CODE REVIEW

**Comment by constantinpape** [comment]
Re deprecation warning: would you be ok with leaving the argument, but not using it at all and raise a warning that states that `storage_options` should be used? I would be on-board with that, but I would rather avoid implementing any consolidation of `storage_options` and chunks.

**Comment by sbesson** [comment]
> Re deprecation warning: would you be ok with leaving the argument, but not using it at all and raise a warning that states that storage_options should be used?

Tricky. When deprecating,I am definitely on board with a loud warning to tell people to upgrade their code, ideally with an idea of when the feature will be dropped. 
In all cases, I would be very cautious about making an existing argument a no-op as that would effectively mean change the behavior of existing code using `chunks` and potentially invalidate data generation.
Actually due to the `**metadata` argument, I am correctly assuming that in the current state of this proposal

```
write_image(image=data, group=root, chunks=chunks, axes="zyx")
```

would use the default chunks but write `chunks` as a key in the `multiscales` metadata?

**Comment by constantinpape** [comment]
Hmm, my problem with this is that the current code is only semi-functional. For example passing chunks via `storage_options` will not `retuple` it and thus likely fail when writing `0.1` and `0.2`.

**Comment by sbesson** [comment]
Definitely worth fixing the retupling of `chunks` via `storage_options` as this is our preferred API moving forward.
Trying to lay down the different scenarios and at least my preference in terms of behavior


```
write_image(image=data, group=root)
```

Expected behavior: unchanged

```
write_image(image=data, group=root, chunks=chunks)
```

Expected behavior: deprecation warning, retuple `chunks` and pass it to `create_dataset`


```
write_image(image=data, group=root, storage_options={"chunks": chunks})
```

Expected behavior: retuple `stoage_options["chunks"]`  and pass it to `create_dataset`


```
write_image(image=data, group=root, chunks=chunks, storage_options={"chunks": chunks})
```

Expected behavior: deprecation warning, ignore `chunks`, retuple `stoage_options["chunks"]` and pass it to `create_dataset`

I would think the only change with your implementation is that https://github.com/ome/ome-zarr-py/pull/184/files#diff-d51ad4cb28d657d5e09c58f40674b95273611983353ab5d7bb2049bbc6303f7bR219 would not unconditionally default to `None` but to the retupled version of `chunks` if it is passed as an explicit keyword argument.

Also happy to hear @joshmoore or @will-moore's opinions on the above

**Comment by constantinpape** [comment]
Ok, I added back the chunk argument in a way that should specify your requirements, @sbesson.

**Comment by constantinpape** [comment]
@sbesson I have added `write_labels` now (and refactored code a bit to reuse the mip creation from `write_image`).
There's also a test for it.

This is good to go from my side, pending the decision on whether we indeed want to remove `chunks` in 0.5; if not the deprecation warning has to be updated.

File: docs/source/python.rst
Comment (sbesson) [inline docs/source/python.rst:38]:
```suggestion
    write_image(image=data, group=root, axes="zyx", storage_options=dict(chunks=(1, size_xy, size_xy)))
```

File: docs/source/python.rst
Comment (constantinpape) [inline docs/source/python.rst:38]: Thanks for catching this

**Review by sbesson — APPROVED** [review]
Nice.

I like the extraction of the pyramid creation logic into a sub-function which clarifies a lot of the contract of `write_image` vs `write_multiscale` (at the code level at least). I assume at some point, `_create_mip` could be turned into a public API for more advanced workflows but that's outside the scope of this PR.

The deprecation of `chunks` makes sense (while keeping the implementation functional) and I think it's fair enough to anticipate and warn we are planning to remove it in the next `0.x.0` release.

No objection from my side to getting this in. I'll leave people mentioned in the description and anyone interested until EOB to express secondary options and merge if no significant objection has been raised. I might follow-up with a quick 

In terms of release, I am away until next Thursday. Anyone interested in carrying out the release early next week? I think it's as simple as updating the CHANGELOG, running `bump2version` and pushing the tag but we might need to sort out permissions and optionally review the instructions.

**Comment by constantinpape** [comment]
> In terms of release, I am away until next Thursday. Anyone interested in carrying out the release early next week? I think it's as simple as updating the CHANGELOG, running `bump2version` and pushing the tag but we might need to sort out permissions and optionally review the instructions.

I could do it on Monday, but I currently don't have any permissions for this repo. But I am also happy to let @will-moore or @joshmoore go ahead with this. (But from my side it would be great to make the release available on Monday since I would like to finish the conversion scripts for Shila's data next week and having this version released would help a lot with this.)

**Comment by will-moore** [comment]
Tested the latest code with https://github.com/ome/omero-cli-zarr/pull/107 and looks good.
Happy to release once this is merged.
@constantinpape could you update the CHANGELOG.md in this PR?

**Comment by constantinpape** [comment]
> Tested the latest code with [ome/omero-cli-zarr#107](https://github.com/ome/omero-cli-zarr/pull/107) and looks good.

Thanks!

> Happy to release once this is merged.
> @constantinpape could you update the CHANGELOG.md in this PR?

I added the changes from my two PRs to the changelog now.

**Comment by joshmoore** [comment]
> I could do it on Monday, but I currently don't have any permissions for this repo. But I am also happy to let @will-moore or @joshmoore go ahead with this.

Also added an ome/ome-zarr-py team with @constantinpape. :+1:

**Comment by constantinpape** [comment]
Thanks for the reviews here and also for adding me to the repo :).
@will-moore: do you think you can make the release today?

**Comment by will-moore** [comment]
Done! https://pypi.org/project/ome-zarr/

**Comment by sbesson** [comment]
Thanks all for driving this

**Pull request merged**
